mne/viz/_brain/_brain.py

```diff
@@ -1463,24 +1458,27 @@ def _configure_shortcuts(self):
             self.plotter.add_key_event(key, partial(func, sign * _ARROW_MOVE))
 
     def _configure_menu(self):
-        # remove default picking menu
-        to_remove = list()
-        for action in self.main_menu.actions():
-            if action.text() == "Tools":
-                to_remove.append(action)
-        for action in to_remove:
-            self.main_menu.removeAction(action)
-
-        # add help menu
-        menu = self.main_menu.addMenu('Help')
-        menu.addAction('Show MNE key bindings\t?', self.help)
+        if self.notebook:
+            return
+        self._renderer._menu_initialize(window=self.window)
+        self._renderer._menu_add_submenu(
+            name="help",
+            desc="Help",
+        )
+        self._renderer._menu_add_button(
+            menu_name="help",
+            name="help",
+            desc="Show MNE key bindings\t?",
+            func=self.help,
+        )
 
     def _configure_status_bar(self):
-        from PyQt5.QtWidgets import QLabel, QProgressBar
-        self.status_msg = QLabel(self.default_status_bar_msg)
-        self.status_progress = QProgressBar()
-        self.status_bar.layout().addWidget(self.status_msg, 1)
-        self.status_bar.layout().addWidget(self.status_progress, 0)
+        if self.notebook:
+            return
+        self._renderer._status_bar_initialize(window=self.window)
+        self.status_msg = self._renderer._status_bar_add_label(
+            self.default_status_bar_msg, stretch=1)
+        self.status_progress = self._renderer._status_bar_add_progress_bar()
```

Comment:
Rather than all this special casing, I would make these part of QtDock / IPythonDock / AbstractDock and just have the IPythonDock be a no-op when necessary

Reply:
(Really the name maybe should be QtGUI / IPythonGUI / AbstractGUI, but the idea is that any special-casing based on backend functionality can be made just based on which backend class is called into rather than using `self.notebook` in `_brain.py`, other than the very first one when you use the `notebook` arg to decide which GUI class to use)

Reply:
It leaves some room for future implementation but at the moment, neither the status bar of the menu bar is implemented for the notebook backend.

Reply:
> It leaves some room for future implementation but at the moment, neither the status bar of the menu bar is implemented for the notebook backend.

Indeed -- the GUI backends will implement things differently. The extreme case of "differently" means "not at all" in some cases, and that's okay.

The more important aspect to me is simplifying code flow for maintenance purposes. Eventually there should be one place in `brain` where `self.notebook` matters, and that's at the beginning. Then everything else that depends on the backend lives in that backend's class. Then people who work on `Brain` don't have to see spaghetti code that sometimes defers to the class and other times uses `self.notebook` to decide what to do. The interface is just that you call to the right method of the GUI wrapper class, whose interface is nicely defined in `_abstract.py`. And if what you need is not there, you modify the abstract + classes as necessary. Make sense?

Reply:
... or to put it in quantitative terms:
```
$ git grep "if self\.notebook"
mne/viz/_brain/_brain.py:        if self.notebook:
mne/viz/_brain/_brain.py:        if self.notebook:
mne/viz/_brain/_brain.py:        if self.notebook:
mne/viz/_brain/_brain.py:        if self.notebook:
mne/viz/_brain/_brain.py:            if self.notebook:
mne/viz/_brain/_brain.py:        if self.notebook:
mne/viz/_brain/_brain.py:        if self.notebook:
mne/viz/_brain/_brain.py:        if self.notebook:
mne/viz/_brain/_brain.py:        if self.notebook:
mne/viz/_brain/_brain.py:            if self.notebook and self._renderer.figure.display is not None:
mne/viz/_brain/callback.py:        if self.notebook:
mne/viz/_brain/callback.py:        if self.notebook:
mne/viz/_brain/mplcanvas.py:        if self.notebook:
mne/viz/_brain/mplcanvas.py:        if self.notebook:
mne/viz/backends/_pyvista.py:        if self.notebook:
```
All of these -- or at least the ones in `_brain`, `callback`, and `mplcanvas` -- should go away, other than one in `_brain.py` during `Brain` instantiation.

If you're convinced, do you want to see how feasible it is to do this in this PR? If not, you or I (if you're not interested in it) could take a stab at it in another one.